components/dashboard/src/admin/UserDetail.tsx

```diff
@@ -0,0 +1,238 @@
+                                    getGitpodService().server.adminSetProfessionalOpenSource(user.id, !isProfessionalOpenSource);
+                                }
+                            }}
+                        >{accountStatement?.subscriptions ? accountStatement.subscriptions.filter(s => Subscription.isActive(s, new Date().toISOString())).map(s => Plans.getById(s.planId)?.name).join(', ') : '---'}</Property>
+                    </div>
+                    <div className="flex w-full mt-6">
+                        <Property name="Feature Flags"
+                            action={{
+                                label: 'Edit Feature Flags',
+                                onClick: () => {
+                                    setEditFeatureFlags(true);
+                                }
+                            }}
+                        >{user.featureFlags?.permanentWSFeatureFlags?.join(', ') || '---'}</Property>
+                        <Property name="Roles"
+                            action={{
+                                label: 'Edit Roles',
+                                onClick: () => {
+                                    setEditRoles(true);
+                                }
+                            }}
+                        >{user.rolesOrPermissions?.join(', ') || '---'}</Property>
+                        <Property name="Student">{isStudent === undefined ? '---' : (isStudent ? 'Enabled' : 'Disabled')}</Property>
+                    </div>
+                </div>
+            </div>
+            <WorkspaceSearch user={user} />
+        </PageWithSubMenu>
+        <Modal visible={editFeatureFlags} onClose={() => setEditFeatureFlags(false)} title="Edit Feature Flags" buttons={[
+            <button className="secondary" onClick={() => setEditFeatureFlags(false)}>Done</button>
+        ]}>
+            <p>Edit feature access by adding or removing feature flags for this user.</p>
+            <div className="flex flex-col">
+                {
+                    flags.map(e => <CheckBox key={e.title} title={e.title} desc="" checked={!!e.checked} onChange={e.onClick} />)
+                }
+            </div>
+        </Modal>
+        <Modal visible={editRoles} onClose={() => setEditRoles(false)} title="Edit Roles" buttons={[
+            <button className="secondary" onClick={() => setEditRoles(false)}>Done</button>
+        ]}>
+            <p>Edit user permissions by adding or removing roles for this user.</p>
+            <div className="flex flex-col">
+                {
+                    rop.map(e => <CheckBox key={e.title} title={e.title} desc="" checked={!!e.checked} onChange={e.onClick} />)
+                }
+            </div>
+        </Modal>
+        <Modal visible={viewAccountStatement} onClose={() => setViewAccountStatement(false)} title="Edit Roles" buttons={[
+            <button className="secondary" onClick={() => setViewAccountStatement(false)}>Done</button>
```

Comment:
**suggestion**: Could we use the same flow we already have in _Integrations_ page? For example, _Permissions_ and _Feature Flags_ could be disabled and only become active when you alter some of the permissions above. It should be fine to ship as is for now.

Currently, the page flashing on the background also slightly degrades the UX of this page when updating roles or feature flags.

Reply:
I didn't introduce a dirty state that would need to be saved. It would introduce some more complexity so I opted for this simple auto-save approach.